jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -181,7 +180,7 @@ public static Content.Source asContentSource(Resource resource, ByteBufferPool b
         Path path = resource.getPath();
         if (path != null)
         {
-            return new ByteChannelContentSource.PathContentSource(new ByteBufferPool.Sized(bufferPool, direct, bufferSize), path, first, length);
+            return Content.Source.from(new ByteBufferPool.Sized(bufferPool, direct, bufferSize), path, first, length);
```

Comment:
There are other places in this class where you could use the new `Content.Source.from()` methods for `ByteBuffer` and `InputStream` types.

And you should use the `ByteBufferPool` which is not currently being used when we create `InputStreamContentSource` here.